IRremoteESP8266.cpp

```diff
@@ -107,6 +107,31 @@ void IRsend::begin() {
 	pinMode(IRpin, OUTPUT);
 }
 
+// Generic method for sending data that is common to most protocols.
+// Default to transmitting the Most Significant Bit (MSB) first.
+void IRsend::sendData(unsigned int onemark, unsigned long onespace,
+                      unsigned int zeromark, unsigned long zerospace,
+                      uint32_t data, uint8_t nbits, bool MSBfirst = true) {
+  if (MSBfirst)  // Send the MSB first.
+    for (uint32_t mask = 1UL << (nbits - 1);  mask;  mask >>= 1)
+      if (data & mask) {  // 1
+        mark(onemark);
+        space(onespace);
+      } else {  // 0
+        mark(zeromark);
+        space(zerospace);
+      }
+  else {  // Send the Least Significant Bit (LSB) first / MSB last.
+    for (uint8_t bit = 0; bit < nbits; bit++, data >>= 1)
+      if (data & B1) {  // 1
```

Comment:
Binary 1. i.e. 0x1. Used to be 100% certain I'm just setting the least significant bit.
It should be that B1 == 0x1 == 1, but I wanted to be architecture/endian neutral.